### DISCUSSION TITLE
setup webhook on pythonanywhere

### DISCUSSION THREAD

**Discussion by Javokhir-1999** [python-telegram-bot/python-telegram-bot#2952] [Q&A]
Hi, I have completed Bot (locally) where I used webhook + `ngrok`  tunnel:

    updater.start_webhook(
		    listen='127.0.0.1',
                port=8443,
	            url_path=settings.TOKEN,
	            key='private.key',
		    # cert='cert.pem',
		    webhook_url = 'https://2068-84-54-80-160.ngrok.io/' + settings.TOKEN,
		) 

All works but, when I moved to  `pythonanywhere` I started getting 404 not found error on `getWebhookInfo` and on `pythonanywhere`  access logs. Here I pinned my new setup and logs on the server:

 I will be glad to have any help

![allgobot py](https://user-images.githubusercontent.com/63791100/163597800-f6f24c8b-61e5-4c34-ac41-28244680c78f.jpg)
![settings py](https://user-images.githubusercontent.com/63791100/163597808-940891d8-ed6b-4abd-962d-3afc2c8d75a6.jpg)
![getWebhookInfo](https://user-images.githubusercontent.com/63791100/163597803-b003fb18-f0c9-4426-bbaf-44896e0a68bc.jpg)
![access log](https://user-images.githubusercontent.com/63791100/163597780-e720cfc6-0f53-4c69-8e3e-f96a1cd78ebe.jpg)
![allgobot run](https://user-images.githubusercontent.com/63791100/163597790-078a3ad3-f28c-4256-b25c-c3171eea94ba.jpg)

**Comment by filiplajszczak** [comment]
You can't run a web app from a console on PythonAnywhere. You need to configure your webhook endpoint as part of your normal web app deployment.

**Reply by askfriends** [reply]
> You can't run a web app from a console on PythonAnywhere. You need to configure your webhook endpoint as part of your normal web app deployment.

so how to do that? sorry for noob question

**Reply by PrometheusPrograms** [reply]
I also have the same issue. I've set up the webhook. I have this code in my VSCode:
@app.route('/update', methods=['POST'])
def webhook():
        if request.method == 'POST':
            repo = git.Repo('./hello-app2')
            origin = repo.remotes.origin
            origin.pull()
            return 'Updated PythonAnywhere successfully', 200
        else:
            return 'Wrong event type', 400

And in github, my webhook URL is configured to be <username>.pythonanywhere.com/update

And I see the log of the webhook being triggered and failing. This is the webhook error message in github:
Content-Length: 207
Content-Type: text/html; charset=utf-8
Date: Thu, 20 Apr 2023 05:54:51 GMT
Server: PythonAnywhere
Body
<!doctype html>
<html lang=en>
<title>404 Not Found</title>
<h1>Not Found</h1>
<p>The requested URL was not found on the server. If you entered the URL manually please check your spelling and try again.</p>
